internal/utils/utils.go

```diff
@@ -32,3 +32,13 @@ func MaxBigInt(a *big.Int, b *big.Int) *big.Int {
 
 	return &max
 }
+
+func Filter[T any](slice []T, predicate func(x T) bool) []T {
```

Comment:
You have some helpers in go libs to play with generics.
https://github.com/formancehq/go-libs/tree/main/collectionutils

Reply:
yeah I'm aware of this lib but for such simple utilities that are basically one-liners it's just better to duplicate them
(I've also tried to avoid this so far because I though it's sort of non-idiomatic in go, but it's just so much clearer this way..)